### PR TITLE
Implement basic Switch

### PR DESCRIPTION
**Pull request by rikner**
Fixes # 

**Type of change:** 

## Motivation (current vs expected behavior)
This is a minimal implementation for a UISwitch-like component

Resources: https://material.io/develop/android/components/switches

Jan-07-2022 15-28-50

now with thumb having a shadow:
switch

## Please check if the PR fulfills these requirements
- [x] Self-review: I am confident this is the simplest and clearest way to achieve the expected behaviour
- [x] There are no dependencies on other PRs or I have linked dependencies through Zenhub
- [x] The commit messages are clean and understandable
- [ ] Tests for the changes have been added (for bug fixes / features)

### CODE REVIEW

File: Sources/Switch.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
+private let frameSize = CGSize(width: 51, height: 21)
+private let thumbSize = CGFloat(31.0)
+private let animationTime = 0.25
+
+private let defaultOnTintColor = CGColor(red: 0.69, green: 0.5, blue: 0.96, alpha: 1)
+private let defaultThumbColor = CGColor(red: 0.30, green: 0, blue: 0.91, alpha: 1)
```
Comment (rikner) [inline Sources/Switch.swift:6]: this could be basically any colours

**Comment by michaelknoch** [comment]
it seems that the switch is invisible on first render and has to be clicked once to make it visible @rikner 
https://beta.record.wtf/play?id=1641471935790_47345

**Comment by rikner** [comment]
> it seems that the switch is invisible on first render and has to be clicked once to make it visible @rikner https://beta.record.wtf/play?id=1641471935790_47345

thx for the hint. fixed it 👍

**Review by michaelknoch — APPROVED** [review]

File: Sources/Switch.swift [hunk 2]
```diff
@@ -0,0 +1,96 @@
+private let frameSize = CGSize(width: 51, height: 21)
+private let thumbSize = CGFloat(31.0)
+private let animationTime = 0.25
+
+private let defaultOnTintColor = CGColor(red: 169 / 255, green: 218 / 255 , blue: 214 / 255, alpha: 1)
+private let defaultThumbColor = CGColor(red: 14 / 255, green: 136 / 255 , blue: 122 / 255, alpha: 1)
+
+private let offThumbColor = CGColor(red: 176 / 255, green: 176 / 255 , blue: 176 / 255, alpha: 1)
+private let offTrackColor = CGColor(red: 225 / 255, green: 225 / 255 , blue: 225 / 255, alpha: 1)
```
Comment (michaelknoch) [inline Sources/Switch.swift:9]: nit: personally I would move these private styling vars to the bottom of the file

**Review by michaelknoch — APPROVED** [review]

**Comment by mouselangelo** [comment]
Hey @rikner - sorry I never got around to reviewing this! I had reviewed the code and wanted to test this locally but just didn't get around to doing that! The implementation looks great though. Thanks 🙏 

I am wondering if adding this to the `samples/getting-started` app, like we did for alerts UI would have made it easier to test. I also like the idea of being able to preview all the components and test them.

I'd even suggest implementing a new `samples/storybook` or `samples/showcase` project - to keep `getting-started` very simple. We could implement something similar to [storybook](https://next--storybookjs.netlify.app/official-storybook), but purely using UIKit. We use storybook in our harmony design system and it's very useful to browse through the various core styles and components.

I'd be happy to support with this. @michaelknoch @ephemer - what do you guys think?

**Pull request merged**